src/target_info.rs

```diff
@@ -85,34 +83,50 @@ fn extract_stack_info(
         let section_range = lowest_address..=highest_address;
         let name = section.name().unwrap_or("<unknown>");
 
-        if active_ram_region.range.contains(&highest_address) {
-            log::debug!(
-                "section `{}` is in RAM at {:#010X} ..= {:#010X}",
-                name,
-                lowest_address,
-                highest_address,
-            );
+        if ram_range.contains(section_range.end()) {
+            log::debug!("section `{}` is in RAM at {:#010X?}", name, section_range);
 
-            if section_range.contains(&(initial_stack_pointer - 1)) {
+            if section_range.contains(stack_range.end()) {
```

Comment:
This changes behavior and makes the logged message below inaccurate

Reply:
Ah, nevermind, `stack_range.end()` is always constant. Perhaps we should just use `let stack_end = initial_stack_pointer - 1;` and only construct the range when returning, that sounds clearer.

Reply:
👍🏾

Reply:
It actually took me quite a while to understand the function, part of it because there were multiple variables representing pretty much the same thing, so I also tried to unify that by only using the ranges.